src/components/tls-checker/index.test.ts

```diff
@@ -0,0 +1,70 @@
+/**********************************************************************************
+ * MIT License                                                                    *
+ *                                                                                *
+ * Copyright (c) 2021 Hyperjump Technology                                        *
+ *                                                                                *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy   *
+ * of this software and associated documentation files (the "Software"), to deal  *
+ * in the Software without restriction, including without limitation the rights   *
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      *
+ * copies of the Software, and to permit persons to whom the Software is          *
+ * furnished to do so, subject to the following conditions:                       *
+ *                                                                                *
+ * The above copyright notice and this permission notice shall be included in all *
+ * copies or substantial portions of the Software.                                *
+ *                                                                                *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    *
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  *
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
+ * SOFTWARE.                                                                      *
+ **********************************************************************************/
+
+import { expect } from '@oclif/test'
+import { checkTLS } from '../tls-checker'
+
+describe('TLS Checker', () => {
+  describe('fail attempt', () => {
+    it('expired domain', async () => {
+      // arrange
+      const url = 'expired.badssl.com'
+
+      try {
+        // act
+        await checkTLS(url)
+      } catch (error) {
+        // assert
+        expect(error.message).to.include(
+          `${url} security certificate has expired at`
+        )
+      }
+    })
+
+    it('tests example.com with long cert expiry threshold', async () => {
+      // arrange
+      const url = 'example.com'
+
+      try {
+        // act
+        await checkTLS(url)
+      } catch (error) {
+        // assert
+        expect(error.message).to.include(
+          `${url} security certificate will expire at`
+        )
+      }
+    })
+  })
+
+  describe('success attempt', () => {
+    it('tests example.com', async () => {
+      // arrange
+      const url = 'example.com'
+
+      // act
+      await checkTLS(url)
```

Comment:
do we expect no error for success case?

Reply:
For now, the function will throw an error if TLS is not valid or pass the threshold, so if it's not thrown, I assume it succeeds. Do you have any suggestions?

Reply:
since this is test, we "need" to make sure the test is passed by expecting something.
Can we put the `checkTLS` result into variable and expect that variable to be null? (for the sake of passing the test)

Reply:
Ok, I got it. I fixed it here [0426c47](https://github.com/hyperjumptech/monika/pull/315/commits/0426c4796cf4765f6bb4b30247340ba6635b2898). Is it enough?